app/src/main/java/org/elasticsearch/node/internal/CrateSettingsPreparer.java

```diff
@@ -71,7 +72,16 @@ public static Environment prepareEnvironment(Settings input, Environment esEnvir
         }
 
         // we put back settings from command line to override the ones from configuration file
-        builder.put(esEnvironment.settings());
+        // but we avoid overriding the cluster name with the default ES cluster name.
+        for (Map.Entry<String, String> entry : esEnvironment.settings().getAsMap().entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            if (!key.equals(ClusterName.CLUSTER_NAME_SETTING.getKey()) ||
+                builder.get(ClusterName.CLUSTER_NAME_SETTING.getKey()) == null ||
+                !value.equals(ClusterName.DEFAULT.value())) {
+                    builder.put(key, value);
+            }
+        }
```

Comment:
this doesn't work because with: builder.put(esEnviroment.settings()) we've already lost the setting from the config file.